rskj-core/src/main/java/co/rsk/rpc/modules/eth/subscribe/SyncNotificationEmitter.java

```diff
@@ -0,0 +1,114 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.rpc.modules.eth.subscribe;
+
+import co.rsk.net.NodeBlockProcessor;
+import co.rsk.rpc.JsonRpcSerializer;
+import com.google.common.annotations.VisibleForTesting;
+import io.netty.channel.Channel;
+import io.netty.handler.codec.http.websocketx.TextWebSocketFrame;
+import org.ethereum.core.Blockchain;
+import org.ethereum.facade.Ethereum;
+import org.ethereum.listener.EthereumListenerAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class SyncNotificationEmitter {
+    private static final Logger logger = LoggerFactory.getLogger(SyncNotificationEmitter.class);
+
+    private final JsonRpcSerializer jsonRpcSerializer;
+    private final NodeBlockProcessor nodeBlockProcessor;
+    private final Blockchain blockchain;
+    private final long startingBlock;
+
+    private final Map<SubscriptionId, Channel> subscriptions = new ConcurrentHashMap<>();
+
+    public SyncNotificationEmitter(
+            Ethereum ethereum,
+            JsonRpcSerializer jsonRpcSerializer,
+            NodeBlockProcessor nodeBlockProcessor,
+            Blockchain blockchain) {
+        this.jsonRpcSerializer = jsonRpcSerializer;
+        this.nodeBlockProcessor = nodeBlockProcessor;
+        this.blockchain = blockchain;
+        this.startingBlock = blockchain.getBestBlock().getNumber();
+        ethereum.addListener(new EthereumListenerAdapter() {
+            @Override
+            public void onLongSyncStarted() {
+                emitSyncStatus(true);
+            }
+
+            @Override
+            public void onLongSyncDone() {
+                emitSyncStatus(false);
+            }
+        });
+    }
+
+    public void subscribe(SubscriptionId subscriptionId, Channel channel) {
+        subscriptions.put(subscriptionId, channel);
+    }
+
+    public boolean unsubscribe(SubscriptionId subscriptionId) {
+        return subscriptions.remove(subscriptionId) != null;
+    }
+
+    public void unsubscribe(Channel channel) {
+        subscriptions.values().removeIf(channel::equals);
+    }
+
+    private synchronized void emitSyncStatus(boolean isSyncing) {
+        if (subscriptions.isEmpty()) {
+            return;
+        }
+        SyncNotification syncNotification = getNotification(isSyncing);
+        subscriptions.forEach((SubscriptionId id, Channel channel) -> {
+            EthSubscriptionNotification request = new EthSubscriptionNotification(
+                    new EthSubscriptionParams(id, syncNotification));
```

Comment:
according to the [docs](https://geth.ethereum.org/docs/rpc/pubsub), `result` could either be `false` or an object with sync statistics. Could you please return `false`, if syncing is not active? Please refer to the examplese of responses below:

```
< {"jsonrpc":"2.0","method":"eth_subscription","params":{"subscription":"0x8a70b5d09d8284c507b1cd1340330f73","result":false}}

< {"jsonrpc":"2.0","method":"eth_subscription","params":{"subscription":"0x8a70b5d09d8284c507b1cd1340330f73","result":{"syncing": true, "status": { "startingBlock": 0, "currentBlock": 0, "highestBlock": 24255}}}
```

Reply:
It is returning false when sync has been stopped/completed. - Please look at line 110 of `SyncNotificationEmitter`. There should be test cases for this as well.

Please let me know if anything else is needed for this.

Reply:
thanks @jamesrowe08 for the corrections you've made.

regarding the format of the response that's being retuned.. what I meant is that your implementation returns something like this:

```
{"jsonrpc":"2.0","method":"eth_subscription","params":{"subscription":"0x8a70b5d09d8284c507b1cd1340330f73","result":{"syncing":false}}}
```
but instead it should be:
```
{"jsonrpc":"2.0","method":"eth_subscription","params":{"subscription":"0x8a70b5d09d8284c507b1cd1340330f73","result":false}}
```
.. where `result` is `false`, not `{"syncing":false}`

Reply:
please refer to the [eth_syncing](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L276) method which does pretty much the same

Reply:
This has been completed. Updated test cases and added additional test case `EthSubscriptionNotificationTest.booleanRequestSync`